examples/util/OpenCVRunner.js

```diff
@@ -0,0 +1,126 @@
+class OpenCVRunner extends BaseRunner {
+  constructor() {
+    super();
+    this._output = null;
+  }
+
+  _getRawModel = async (url) => {
+    let status = 'ERROR';
+
+    if (url !== undefined) {
+      const arrayBuffer = await this._loadURL(url, this._progressHandler, true);
+      const bytes = new Uint8Array(arrayBuffer);
+      switch (url.split('.').pop()) {
+        case 'onnx':
+          const err = onnx.ModelProto.verify(bytes);
+          if (err) {
+            throw new Error(`The model file ${url} is invalid, ${err}`);
+          }
+          try {
+            cv.FS_createDataFile('/', this._currentModelInfo.modelId, bytes, true, false, false);
+          } catch (e) {
+            if (e.errno === 17) {
+              console.log(`${this._currentModelInfo.modelId} already exited.`);
+              status = 'SUCCESS'
+            } else {
+              console.error(e);
+            }
+          }
+          break;
+        default:
+          throw new Error(`Unrecognized model format, support TFLite | ONNX | OpenVINO model`);
+      }
+    } else {
+      throw new Error(`There's none model file info, please check config info of modelZoo.`);
+    }
+
+    this._setLoadedFlag(true);
+    return status;
+  };
+
+  _getOtherResources = async () => {
+    // Override by inherited if needed, likes load labels file
+  };
+
+  _getModelResources = async () => {
+    await this._getRawModel(this._currentModelInfo.modelFile);
+    await this._getOtherResources();
+  };
+
+  loadModel = async (modelInfo) => {
+    if (this._bLoaded && this._currentModelInfo.modelFile === modelInfo.modelFile) {
+      return 'LOADED';
+    }
+
+    // reset all states
+    this._setLoadedFlag(false);
+    this._setInitializedFlag(false);
+    this._setModelInfo(modelInfo);
+    await this._getModelResources();
+  };
+
+  compileModel = (options) => {
+    if (!this._bLoaded) {
+      return 'NOT_LOADED';
+    }
+
+    if (this._bInitialized) {
+      return 'INITIALIZED';
+    }
+
+    this._setInitializedFlag(false);
+    let model = null;
+
+    switch (this._currentModelInfo.format) {
+      case 'ONNX':
+        model = cv.readNetFromONNX(this._currentModelInfo.modelId);
+        break;
+      default:
+        throw new Error(`Unsupported '${this._currentModelInfo.format}' input`);
+    }
+
+    this._setModel(model);
+    this._setInitializedFlag(true);
+    return 'SUCCESS';
+  };
+
+  _getInputTensor = (src, options) => {
+    const mean = options.preOptions.mean;
+    const std = options.preOptions.std;
+    const [width, height, channel] = options.inputSize;
+    let mat = cv.imread(src.id);
```

Comment:
does it support video stream?

Reply:
No, imread interface only reads image or canvas. 
And OpenCV.js provides VideoCapture interface to capture live stream with camera, so I'll use cv.VideoCapture to do image classification with Live Camera inputs.

Reply:
[cv.VideoCapture](https://github.com/opencv/opencv/blob/master/modules/js/src/helpers.js#L118) and [cv.imread](https://github.com/opencv/opencv/blob/master/modules/js/src/helpers.js#L45) are just simple helpers. 

Probably, we can reuse existing example code to get data from image or video. I guess the only difference between each framework is how to pupulate their inputs with the data, for OpenCV.js, it is cv.Mat.

What do you think?

Reply:
Can we support video stream in this PR?

Reply:
Thanks for your suggestion. I will investigate how to converting input data for setInput method of net for OpenCV.js, then support video stream in this PR.